control/Director.php

```diff
@@ -428,36 +443,56 @@ public static function set_current_page($page) {
 
 	/**
 	 * Turns the given URL into an absolute URL.
-	 * By default non-site root relative urls will be evaluated relative to the current request.
+	 * By default non-site root relative urls will be evaluated relative to the current base_url
 	 *
 	 * @param string $url URL To transform to absolute
-	 * @param bool $relativeToSiteBase Flag indicating if non-site root relative urls should be
-	 * evaluated relative to the site BaseURL instead of the current url.
+	 * @param string $relativeParent Method to use for evaluating relative urls.
+	 * Either one of BASE (baseurl), ROOT (site root), or REQUEST (requested page).
+	 * Defaults to BASE, which is the same behaviour as template url resolution.
+	 * Igored if the url is absolute or site root.
 	 * @return string The fully qualified URL
 	 */
-	public static function absoluteURL($url, $relativeToSiteBase = false) {
-		if(!isset($_SERVER['REQUEST_URI'])) return false;
+	public static function absoluteURL($url, $relativeParent = self::BASE) {
+		if(is_bool($relativeParent)) {
+			// Deprecate old boolean second parameter
+			Deprecation::notice('5.0', 'Director::absoluteURL takes an explicit parent for relative url');
```

Comment:
I assume we'll include this in the 5.0 changelog once that release is prepped, as part of going through all deprecation notices? Might be a bit premature to create a 5.0 changelog already, but it'd help to describe the change while somebody can still remember making it ;)